src/EditorFeatures/Core/Implementation/CodeFixes/CodeFixService.cs

```diff
@@ -651,17 +669,23 @@ private static ImmutableArray<string> GetAndTestFixableDiagnosticIds(CodeFixProv
             return fixerMap;
         }
 
-        private static ImmutableDictionary<LanguageKind, Lazy<ISuppressionFixProvider>> GetSuppressionProvidersPerLanguageMap(
-            Dictionary<LanguageKind, List<Lazy<ISuppressionFixProvider, CodeChangeProviderMetadata>>> suppressionProvidersPerLanguage)
+        private static ImmutableDictionary<LanguageKind, Lazy<ImmutableArray<ISuppressionOrConfigurationFixProvider>>> GetsuppressionOrConfigurationProvidersPerLanguageMap(
+            Dictionary<LanguageKind, List<Lazy<ISuppressionOrConfigurationFixProvider, CodeChangeProviderMetadata>>> suppressionOrConfigurationProvidersPerLanguage)
         {
-            var suppressionFixerMap = ImmutableDictionary.Create<LanguageKind, Lazy<ISuppressionFixProvider>>();
-            foreach (var languageKindAndFixers in suppressionProvidersPerLanguage)
+            var suppressionOrConfigurationFixerMap = ImmutableDictionary.Create<LanguageKind, Lazy<ImmutableArray<ISuppressionOrConfigurationFixProvider>>>();
+            foreach (var languageKindAndFixers in suppressionOrConfigurationProvidersPerLanguage)
             {
-                var suppressionFixerLazyMap = new Lazy<ISuppressionFixProvider>(() => languageKindAndFixers.Value.SingleOrDefault().Value);
-                suppressionFixerMap = suppressionFixerMap.Add(languageKindAndFixers.Key, suppressionFixerLazyMap);
+                var builder = ArrayBuilder<ISuppressionOrConfigurationFixProvider>.GetInstance();
+                foreach (var languageKindAndFixersValue in languageKindAndFixers.Value)
+                {
+                    builder.Add(languageKindAndFixersValue.Value);
+                }
+
+                var lazySuppressionOrConfigurationFixers = new Lazy<ImmutableArray<ISuppressionOrConfigurationFixProvider>>(() => builder.ToImmutableAndFree());
```

Comment:
i don't get how this Lazy helps out here.  You've done all the work *prior* to instantiating it.  so might as well just pass along the array.  If this really does need to be lazy, theabove work should move into the lambda.